networkx/algorithms/similarity.py

```diff
@@ -1642,6 +1643,170 @@ def panther_similarity(
     return top_k_with_val
 
 
+@nx._dispatchable(edge_attrs="weight")
+def panther_vector_similarity(
+    G, source, D=50, k=5, path_length=5, c=0.5, delta=0.1, eps=None, weight="weight"
```

Comment:
Let's make these parameters keyword only -- since this is a new function that will lead to easier to read code using this. 
```suggestion
    G, source, *, D=50, k=5, path_length=5, c=0.5, delta=0.1, eps=None, weight="weight"
```